core/types/rollup_cost.go

```diff
@@ -308,6 +318,46 @@ func newL1CostFuncEcotone(l1BaseFee, l1BlobBaseFee, l1BaseFeeScalar, l1BlobBaseF
 	}
 }
 
+// NewTotalRollupCostFunc return a TotalRollupCostFunc that computes the total rollup cost, consisting
+// of both, the data availability cost and the operator cost.
+func NewTotalRollupCostFunc(config *params.ChainConfig, statedb StateGetter) TotalRollupCostFunc {
+	if !config.IsOptimism() {
+		return nil
+	}
+	l1CostFunc := NewL1CostFunc(config, statedb)
+	operatorCostFunc := NewOperatorCostFunc(config, statedb)
+
+	return func(tx RollupTransaction, blockTime uint64) *uint256.Int {
+		// proper caching is happening inside the individual cost functions
+		l1Cost := l1CostFunc(tx.RollupCostData(), blockTime)
+		operatorCost := operatorCostFunc(tx.Gas(), blockTime)
+		if l1Cost == nil && operatorCost == nil {
+			return nil
+		}
+
+		var totalCost *uint256.Int
+		var overflow bool
+		if l1Cost != nil {
+			totalCost, overflow = uint256.FromBig(l1Cost)
+			if overflow {
+				panic("overflow in total rollup cost: l1Cost")
+			}
+		} else {
+			totalCost = new(uint256.Int)
+		}
```

Comment:
Nit: I find this a little surprising to read. Why not just create `totalCost` on L338 and conditionally add to it on L341? That harmonizes more with what you do below anyway.

Reply:
It saves one allocation and is otherwise equivalent.

I read it like "if A then initiate this way, else initiate that way"